src/main/java/com/wojtek/parkingmeter/controllers/TicketController.java

```diff
@@ -1,29 +1,86 @@
 package com.wojtek.parkingmeter.controllers;
 
-import com.wojtek.parkingmeter.repositories.TicketRepository;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RestController;
+import com.wojtek.parkingmeter.helpers.*;
+import com.wojtek.parkingmeter.helpers.enums.HasStartedEnum;
+import com.wojtek.parkingmeter.model.TicketDTO;
+import com.wojtek.parkingmeter.services.TicketService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.NoSuchElementException;
 
 @RestController
 public class TicketController {
 
+    @Autowired
+    JdbcTemplate jdbcTemplate;
 
-    private final TicketRepository ticketRepository;
+    private final TicketService ticketService;
 
-    public TicketController(TicketRepository ticketRepository) {
-        this.ticketRepository = ticketRepository;
+    public TicketController(TicketService ticketService) {
+        this.ticketService = ticketService;
     }
 
-    @GetMapping("/{id}/start")
-    public void startTicket(@PathVariable Long id){return;}
+    @GetMapping("/start/{ticket_type}/{nr_plate}")
+    public ResponseEntity<TicketDTO> startTicket(@PathVariable String ticket_type, @PathVariable String nr_plate) {
+
+
+        if (ticketService.hasStarted(nr_plate).getHasStarted().equals(HasStartedEnum.YES))
+            return ResponseEntity.status(HttpStatus.IM_USED).body(new TicketDTO()); // tutaj bym zwrócił info że takkie auto ma już bilet. Użyłem HttpStatus.IM_USED, ale chyba to nie do tego.
+        if (Validator.validateNewTicket(ticket_type, nr_plate))
+            return ResponseEntity.ok().body(ticketService.startTicket(ticket_type, nr_plate));
+        else
+            return ResponseEntity.badRequest().body(new TicketDTO());
+
+    }
 
-    @GetMapping("/{id}/stop")
-    public void stopTicket(@PathVariable Long id){return;}
+    @GetMapping("/stop/{id}")
+    public ResponseEntity<String> stopTicket(@PathVariable Long id) {
+
+        if (!Validator.checkIfAlreadyStarted(jdbcTemplate, id))
+            return ResponseEntity.ok().body("TICKET ALREADY STOPPED");
```

Comment:
czy zwracanie takiego body to dobry pomysł? co się stanie jak będziesz chciał zwrócić więcej informacji?